xonsh/environ.py

```diff
@@ -7,72 +7,153 @@
 import builtins
 import platform
 import subprocess
+from datetime import datetime
+from functools import partial
 from warnings import warn
+from collections import MutableMapping
 
 from xonsh import __version__ as XONSH_VERSION
 from xonsh.tools import TERM_COLORS
 
-def current_branch(cwd=None):
-    """Gets the branch for a current working directory. Returns None
-    if the cwd is not a repository.  This currently only works for git, 
-    bust should be extended in the future.
+# This is only needed because of the way that we create prompt objects in
+# shell.py.  If we find a way to create prompt objects before
+# environ.add_prompt_var() can be called then we wouldn't need this.
+_prompt_vars_queue = {}
```

Comment:
It's not a constant.  It's being modified during runtime.

It can't be an instance variable because we may not have an instance when the toplevel add_prompt_vars() function that @wrywerytwreywery wanted to have is called.